libcodechecker/cmd/cmd_line_client.py

```diff
@@ -321,7 +326,61 @@ def getDiffReportDir(client, baseids, report_dir, diff_type):
                     filtered_reports.append(result)
         return filtered_reports
 
-    def printReports(client, reports, output_format):
+    def cache_report_file(file_cache, file_id):
+        if file_id not in file_cache:
+            source = client.getSourceFileData(file_id, True,
+                                              ttypes.Encoding.BASE64)
+            file_content = base64.b64decode(source.fileContent)
+            file_cache[file_id] = {'id': file_id,
+                                   'path': source.filePath,
+                                   'content': file_content}
+
+        return file_cache[file_id]
+
+    def get_report_data(client, reports, file_cache):
+        file_sources = {}
+        report_data = []
+
+        for report in reports:
+            file_sources[report.fileId] = cache_report_file(file_cache,
+                                                            report.fileId)
+            details = client.getReportDetails(report.reportId)
+            events = []
+            for index, event in enumerate(details.pathEvents):
+                file_sources[event.fileId] = cache_report_file(file_cache,
+                                                               event.fileId)
+                events.append({'line': event.startLine,
+                               'col': event.startCol,
+                               'file': event.fileId,
+                               'msg': event.msg,
+                               'step': index + 1})
+            report_data.append(events)
+
+        return {'files': file_sources,
+                'reports': report_data}
+
+    def report_to_html(client, reports, diff_type):
+        html_builder = PlistToHtml.HtmlBuilder(context.path_plist_to_html_dist)
+
+        file_report_map = defaultdict(list)
+        for report in reports:
+            file_report_map[report.fileId].append(report)
+
+        file_cache = {}
+        for file_id, file_reports in file_report_map.iteritems():
+            checked_file = file_reports[0].checkedFile
+            filename = os.path.basename(checked_file)
+
+            report_data = get_report_data(client, file_reports, file_cache)
+
+            output_path = os.path.join(args.export_dir,
+                                       filename + '_' + str(file_id) +
+                                       '_' + diff_type + '.html')
```

Comment:
I'm not sure we should add the diff type to the filename.
I thought for each diff type I would create a separate export folder `-e new_reports` ...